// dao/area.js
var ObjectId = require('mongodb').ObjectID;
var dbConn = require('dao/connexion');
var coleccionSelecc2 = "areas";

module.exports = {
	insertar: function(docEntidad,callback){
		dbConn.conectar(function(db){
			db.collection(coleccionSelecc).
			insert(docEntidad, function(err, resultado) {
    			if(err) throw err;
    			callback(resultado);
  			});      
		});
	},
	borrar: function(unId,callback){
		dbConn.conectar(function(db){
			db.collection(coleccionSelecc2).
			remove({"_id":new ObjectId(unId)}, function(err, resultado) {
    			if(err) throw err;
    			callback(resultado);
  			});      
		});
	},
	actualizar: function(docEntidad,callback){
		dbConn.conectar(function(db){
			console.log("entro")
			var id = docEntidad._id;
			console.log("docL"+docEntidad._id);
			console.log("IDI= "+id);
			db.collection(coleccionSelecc).
			update({"_id" : new ObjectId(id)},
				{"nombre":docEntidad.nombre,
				"correo":docEntidad.correo,
				"interno":docEntidad.interno},
				 function(err, resultado) {
    			if(err) throw err;
    			callback(resultado);
  			});      
		});
	},
	buscar: function(idDocEntidad,callback){
		dbConn.conectar(function(db){
			console.log(idDocEntidad);
			console.log(coleccionSelecc2);
			db.collection(coleccionSelecc2)
			.findOne({ "_id" : new ObjectId(idDocEntidad) }, function(err,resultado) {
    			if(err) throw err;
    			console.log("--> "+resultado);
    			callback(resultado);
  			});   
		});
	},
	buscarTodos: function(callback){
	
		dbConn.conectar(function(db){
			db.collection(coleccionSelecc2).
			find({}).toArray(function(err, resultado) {
    			if(err) throw err;
    			console.log(resultado);
    			callback(resultado);

  			});      
		});
	}
}
