// dao/expediente.js
var ObjectId = require('mongodb').ObjectID;
var dbConn = require('dao/connexion');
var coleccionSelecc = "expedientes";

module.exports = {
	insertar: function(docEntidad,callback){
		dbConn.conectar(function(db){
			db.collection(coleccionSelecc).
			insert(docEntidad, function(err, resultado) {
    			if(err) throw err;
    			callback(resultado);
  			});      
		});
	},
	borrar: function(unId,callback){
		dbConn.conectar(function(db){
			db.collection(coleccionSelecc).
			remove({"_id":new ObjectId(unId)}, function(err, resultado) {
    			if(err) throw err;
    			callback(resultado);
  			});      
		});
	},
	actualizar: function(docEntidad,callback){
		dbConn.conectar(function(db){
			var id = docEntidad._id;
			console.log("docL"+docEntidad._id);
			console.log("IDI= "+id);
			db.collection(coleccionSelecc).
			update({"_id" : new ObjectId(id)},docEntidad, function(err, resultado) {
    			if(err) throw err;
    			callback(resultado);
  			});      
		});
	},
	buscar: function(idDocEntidad,callback){
		dbConn.conectar(function(db){
			console.log(idDocEntidad);
			console.log(coleccionSelecc);
			db.collection(coleccionSelecc)
			.findOne({ "_id" : new ObjectId(idDocEntidad) }, function(err,resultado) {
    			if(err) throw err;
    			console.log("--> "+resultado);
    			callback(resultado);
  			});   
		});
	},
	buscarTodos: function(callback){
		dbConn.conectar(function(db){
			db.collection(coleccionSelecc).
			find({}).toArray(function(err, resultado) {
    			if(err) throw err;
    			console.log(resultado);
    			callback(resultado);
  			});      
		});
	}
}
